go-controller/pkg/ovn/policy.go

```diff
@@ -93,6 +96,57 @@ func (oc *Controller) syncNetworkPolicies(networkPolicies []interface{}) {
 	}
 }
 
+func (oc *Controller) getNetworkPoliciesLocked() map[types.NamespacedName]*networkPolicy {
+	//Get lock on controller's networkPolicies struct
+	oc.networkPoliciesMutex.Lock()
+
+	npInfo := oc.networkPolicies
+
+	if npInfo == nil {
+		return nil
+	}
+
+	// Check that the networkPolicy wasn't deleted while we were waiting for the lock
+	//Don't need this here
+	//oc.networkPoliciesMutex.Lock()
+	//defer oc.networkPoliciesMutex.Unlock()
+	//if reflect.DeepEqual(npInfo,oc.networkPolicies) {
+	//	npInfo.Unlock()
+	//	return nil
+	//}
+
+	return npInfo
+
+}
+
+// waitForNamespaceLocked waits up to 10 seconds for a Namespace to be known; use this
+// rather than getNamespaceLocked when calling from a thread where you might be processing
+// an event in a namespace before the Namespace factory thread has processed the Namespace
+// addition.
+func (oc *Controller) waitForNetworkPoliciesLocked(ns string) (map[types.NamespacedName]*networkPolicy, error) {
+	var npInfo map[types.NamespacedName]*networkPolicy
+
+	//Ensure namespace is created before applying policies
+	//Do we need this? still requires locking on NS creation
```

Comment:
it doesn't seem necessary if you decouple network policy from namespace

Reply:
Sounds good, I just worry about condition where a network policy could be applied before the namespace in some scenarios